modules/querier/http.go

```diff
@@ -296,6 +297,78 @@ func (q *Querier) SpanMetricsSummaryHandler(w http.ResponseWriter, r *http.Reque
 	w.Header().Set(api.HeaderContentType, api.HeaderAcceptJSON)
 }
 
+func (q *Querier) QueryRangeHandler(w http.ResponseWriter, r *http.Request) {
+	var (
+		err  error
+		resp *tempopb.QueryRangeResponse
+	)
+
+	// Enforce the query timeout while querying backends
+	ctx, cancel := context.WithDeadline(r.Context(), time.Now().Add(q.cfg.Search.QueryTimeout))
+	defer cancel()
+
+	span, ctx := opentracing.StartSpanFromContext(ctx, "Querier.QueryRangeHandler")
+	defer span.Finish()
+
+	errHandler := func(ctx context.Context, span opentracing.Span, err error) {
+		if errors.Is(err, context.Canceled) {
+			// todo: context is also canceled when we hit the query timeout. research what the behavior is
+			// ignore this error. we regularly cancel context once queries are complete
+			span.SetTag("error", err.Error())
+			return
+		}
+
+		if ctx.Err() != nil {
+			span.SetTag("error", ctx.Err())
+			return
+		}
+
+		if err != nil {
+			span.SetTag("error", err.Error())
+		}
+	}
+
+	defer func() {
+		errHandler(ctx, span, err)
+		m := jsonpb.Marshaler{}
+
+		jsBytes, funcErr := m.MarshalToString(resp)
```

Comment:
Testing out the error handling in local blocks processor and it leads to nil resp here triggers the error `Marshal called with nil`.

Reply:
Updated the error handling here.  Good call out.